### ISSUE TITLE
Docs: ban-ts-comment says only @ts-check is allowed by default, but @ts-expect-error is too

### ISSUE THREAD

**Issue by zaewc** [typescript-eslint/typescript-eslint#12480]
### Before You File a Documentation Request Please Confirm You Have Done The Following...

- [x] I have looked for existing [open or closed documentation requests](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Adocumentation) that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.
- [x] If I used AI assistance to prepare this issue, I have reviewed and followed the [AI Contribution Policy](https://typescript-eslint.io/contributing/ai-policy/).

### Suggested Changes

The ban-ts-comment docs currently say

> By default, only `@ts-check` is allowed, as it enables rather than suppresses errors.

This seems a bit misleading, since the default config also allows `@ts-expect-error` when it has a description

```
'ts-check': false,
'ts-expect-error': 'allow-with-description',
'ts-ignore': true,
'ts-nocheck': true,
```

Maybe the sentence could be updated to

> By default, `@ts-check` is allowed, and `@ts-expect-error` is allowed when it includes a description. `@ts-ignore` and `@ts-nocheck` are reported.

### Affected URL(s)

https://typescript-eslint.io/rules/ban-ts-comment

### Additional Info

This is separate from #12464. That one was about the `ts-expect-error` schema description

**Comment by StyleShit** [comment]
Yes, I agree that this is slightly misleading

I think we should keep the "... as it enables rather than suppresses errors." part, though
